### ISSUE TITLE
How to Train 2D Detector

### ISSUE THREAD

**Issue by jn-z** [Zengyi-Qin/MonoGRNet#9]
I really appreciate your awesome work.
I have one questions about training schedule.

As your paper said,
"We first train the 2D detector, along with the backbone, for 120K iterations using the Adam optimizer . Then ...."

For the first training stage for training " first train the 2D detector, along with the backbone, for 120K iterations using the Adam optimizer ", How to set the flag

train_2d = True ?

**Comment by Zengyi-Qin** [comment]
In both include/tensorvision.py and decoder/fastBox.py, set joint_2d_and_3d = True and others = False. The 2D detection model is not trained using this code. It's trained on the official KittiBox framework https://github.com/MarvinTeichmann/KittiBox. This repo supports training the 3d and 2d parts together but not the 2d only.

**Comment by jn-z** [comment]
Thank you for your reply.

Does this mean that I can train 3D detectors directly with pre-trained 2D detector weights?

As your paper said,　I set  joint_3d = true and others = false to train 3D detectors, and then set joint_2d_3d = true and others = false to train both 2D and 3D detectors in the third training stage. But the effect is not very good.  In addition, I found that the effect of the 2D detector will be reduced in the third training stage.  Is there a problem with my training method?
